lib/engine/game/g_18_cuba/game.rb

```diff
@@ -0,0 +1,216 @@
+                      },
+                    ],
+                  },
+                  {
+                    name: '4',
+                    distance: 4,
+                    price: 300,
+                    rusts_on: '8',
+                    variants: [
+                      {
+                        name: '4+',
+                        distance: 4,
+                        price: 340,
+                      },
+                    ],
+                  },
+                  {
+                    name: '5',
+                    distance: 5,
+                    price: 500,
+                    variants: [
+                      {
+                        name: '5+',
+                        distance: 5,
+                        price: 550,
+                      },
+                    ],
+                  },
+                  {
+                    name: '6',
+                    distance: 6,
+                    price: 600,
+                    variants: [
+                      {
+                        name: '6+',
+                        distance: 6,
+                        price: 660,
+                      },
+                    ],
+                  },
+                  {
+                    name: '8',
+                    distance: 8,
+                    price: 700,
+                    variants: [
+                      {
+                        name: '4D',
+                        distance: 4,
+                        price: 800,
+                      },
+                    ],
+                  }].freeze
+
+        def operating_round(round_num)
+          Round::Operating.new(self, [
+            Engine::Step::Bankrupt,
+            Engine::Step::Exchange,
+            Engine::Step::SpecialTrack,
+            Engine::Step::SpecialToken,
+            Engine::Step::BuyCompany,
+            Engine::Step::HomeToken,
+            Engine::Step::Track,
+            Engine::Step::Token,
+            Engine::Step::Route,
+            Engine::Step::Dividend,
+            Engine::Step::DiscardTrain,
+            Engine::Step::BuyTrain,
+            [Engine::Step::BuyCompany, { blocks: true }],
+          ], round_num: round_num)
+        end
+
+        def init_stock_market
+          StockMarket.new(self.class::MARKET, [], zigzag: :flip)
+        end
+
+        def multiple_buy_only_from_market?
+          !optional_rules&.include?(:multiple_brown_from_ipo)
+        end
+
+        def num_trains(train)
+          num_players = [@players.size, 2].max
+          TRAIN_FOR_PLAYER_COUNT[num_players][train[:name].to_sym]
+        end
+
+        def company_header(company)
+          company.id[0] == self.class::COMPANY_CONCESSION_PREFIX ? 'CONCESSION' : 'COMMISSIONER'
+        end
+
+        def setup
+          super
+          @tile_groups = init_tile_groups
+          update_opposites
+        end
+
+        def init_tile_groups
+          self.class::TILE_GROUPS
+        end
+
+        def update_opposites
```

Comment:
You should remove this method, and add `include DoubleSidedTiles` to use the methods in [double_sided_tiles.rb](https://github.com/tobymao/18xx/blob/087b3018ee5b8e844a9518c1dc5401637d2418ab/lib/engine/game/double_sided_tiles.rb#L8)